### PR TITLE
Update the OpenShift guide to version 4.12 and use OpenShift Serverless

### PR DESCRIPTION
**Pull request by ReToCode**
**What this PR does / why we need it**:
The current [OpenShift documentation](https://github.com/kserve/kserve/blob/master/docs/OPENSHIFT_GUIDE.md) is outdated and is using some bundled components from OpenShift (like Service Mesh) and some upstream resources (like Knative OSS).

**Changes**
- The OpenShift documentation was updated
- The documentation refers to the two options as an ingress layer
- The documentation now contains the necessary resources to install KServe on OCP

**Feature/Issue validation/testing**:
The new guides were tested on OCP 4.12.

**Notes**
For these guides to work, additional PRs are needed. Some of them are already merged, others are in review:
- https://github.com/kserve/kserve/pull/2853
- https://github.com/kserve/kserve/pull/2839
- https://github.com/kserve/kserve/pull/2845

**Release note**:
NONE

### CODE REVIEW

**Comment by ReToCode** [comment]
FYI @skonto

File: docs/openshift/serverless/gateways.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
...a2d0Ckh4cDhsQ3lkSUFaYVkzYXUweGRvMXlXNEV0UDVpNjEySk1kYm9TQmhpbGJJQmFpRjRiZ3AzRjA3WFJQeWJBRFgKS1c2aitvTW5SMlRwd1ZsUmJIeW1OT1FYT1hOUEhRWForUFM2bDVjaWs1QlVRVi8vL2NySUFvS3pIUHpOckl3OQpZMGZoWmNrZHRpL0pIZVJZandnWCsvb3I2UUtCZ1FEZlF2ME1ld01mQ3JmWVRCdlVGZldncm9pbU9PWEFhV2M4ClVBVVB2MXdQYjVSbkY5eWdDQjRQL2l5QVlwOWl1aHBodDM3NjdVcGVsL3pCQitmNzZ0bHZaNWIySzBDWHNwaUIKdGNSSnU2ZWZLdEFqZGNqQ3BRM3hpelpTNjNFSHJIWDREY2s4QlhVcG96Q3R5N1BPeXZobDRwMFdyL2gwWW1OYgplU0FWcFViTG1RS0JnUURLTHFIUmwyKzI1WDRZcW45OGIvWXVsbEFjSFlyYXNaVldDNkdWdDZ5enJlKzlTSzBnCklqaUJHK3pGSkFEQkQ2Y0s4WTRWMGRqMTZ2UmNXalBmZ2VPa0taTU9OODU0VEtRWEZDNmNqQjJWY3htRzdrQW8KWDJRazRQa21IZWZna3dMVXV5NW5KeXQ0Q2U3blZDTGwyWTRMTk5IOXQ5b0puS25KdU91MmZCcGNrUUtCZ0F6OApkSU9aVkNFbUduTjJXZGdJUHZWTnNaMFppaU9hL2VwQUxVc3hNa1dqazlvN1JSWDU0dVhEUHd0b3NTU285b2ZnCmlINUg5eDl4Yjc0Nm0zL0h0VVlKbkhwTkljQ3hIclhNd05JWkhETGg1cUZwWkhnTjZiVzNCejNqZS91YVNISloKT3U5RzBmM09CRExYdW1tNDNLSHdnSHFsV2FwTFhzUWZVNEp1enFOaEFvR0FYRGhoaFBZTWt3TFhzQUdxSmlpOApUcnd3cGExN21vbzJoQ2FncFRsemdaR281bjZFandoM1ZoN0JaQ0hJc0FmY3ZoUUZ2OE1UZXVGQzBvNUZGU3gwCjN0MHpKQWdrWFBXYnhXQU9mU3FZc1NGR0tyVHFuNngzcTRWQXVyZ09Qa2pFd3VIUkFMajJyMFZVWUg4OWRab0QKWDBGcUM4SUNRWjRlbDFVd0c5bXAvaEE9Ci0tLS0tRU5EIFBSSVZBVEUgS0VZLS0tLS0K
```
Comment (yuzisun) [inline docs/openshift/serverless/gateways.yaml:56]: Is this a security concern to leak the private key ?

Comment (ReToCode) [reply docs/openshift/serverless/gateways.yaml:56]: It is a self-signed example.com certificate. As it is necessary for a correct setup, I added it here with a note that it should be changed/updated depending on the installation context.

**Comment by yuzisun** [comment]
Thanks @ReToCode !
/lgtm
/approve

**Pull request merged**